### DISCUSSION TITLE
Implement soft-delete

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#4113] [Ideas]
With the current setup it's easy to inadvertently erase a lot of records by mistake.

We should show a toast that defers sending the deletion request by a few seconds. 

The deletion shouldn't still happen if I move to a new page.

See prototype here: https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=5215-59367&mode=design&t=3yzJPjValanNrceO-4

**Comment by Bonapara** [comment]
Design ready whenever needed

![Image](https://github.com/twentyhq/twenty/assets/19412894/43e8dd79-4bd1-4a29-9c6a-410666e63e20)


https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=5215-59367&mode=design&t=3yzJPjValanNrceO-4

**Comment by FelixMalfait** [comment]
Great! This issue initially referred to frontend + backend work (deleted_at field + ability to recover at any point in time). I think it's good enough to solve the solution with your design (delaying the request), so I'm not creating a new issue related to the backend work at this point, we'll probably have to handle it some day

**Comment by gitstart-app** [comment]
Here is the GitStart Ticket for this issue: https://clients.gitstart.com/twenty/5449/tickets/TWNTY-265

**Comment by FelixMalfait** [comment]
Note: the hook for toasts is called useSnackBar

**Comment by charlesBochet** [comment]
Posting it here too:
@gitstart-twenty After many discussions, we have decided that this was not the right approach.
We are willing to put soft deletion into place, and to give the ability to restore entities instead. I will close this pull-request, thank you for working on that, it has raised interesting discussions internally, and sorry for the waste of time on your side :(

**Comment by abhithory** [comment]
@FelixMalfait , it seems very interesting to me. Can i work on this?

**Comment by FelixMalfait** [comment]
@abhithory this one is more difficult than I initially thought. I wanted to do it on the frontend initially. But in the end we need a global strategy for soft-delete on all objects

**Comment by abhithory** [comment]
Ok @FelixMalfait , I am interested in this and can work if you can guide me little bit

**Comment by charlesBochet** [comment]
@Weiko I remember we discussed this one together. It would be great to write our discussion here to provide as much context as possible

**Comment by FelixMalfait** [comment]
It's a hard problem to build soft-delete well (how to cascade deletion properly, apply the scope on pg_graphql queries etc.)
Probably best to invest in logs as a first step. Converting to discussion

**Comment by BOHEUS** [comment]
Soft-delete is available for a long time now
